cpp/include/cuspatial/detail/utility/device_atomics.cuh

```diff
@@ -267,20 +267,20 @@ __device__ inline float atomicMax(thrust::device_ptr<float> ptr, float val)
 /**
  * @brief Factory method to create atomic_ref from a thrust::device_reference
  */
-template <typename T>
+template <typename T, cuda::thread_scope Scope = cuda::thread_scope_device>
```

Comment:
So the default here is different from the default used by `cuda::atomic_ref`. This may cause surprise, so you should at least mention this in an `@note`.

Reply:
Ah I didn't notice, I'd rather keep it the same as cuda and be explicit in user code.

Reply:
I find it putting `Scope` as the second parameter here can make the compiler confused about the deduced type of `T` since I need to explicitly specify the type. Therefore I switched the order of `Scope` and `T` and always require user to explicitly specify thread scope.